File: kid-with-greatest-number-of-candies/kid-with-greatest-number-of-candies.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function(candies, extraCandies) {
    var truthValues = []

    const maxCandies = Math.max(...candies)

    for (const candy of candies) {
        if (candy + extraCandies >= maxCandies) {
            truthValues.push(true)
        } else {
            truthValues.push(false)
        }
    }

    return truthValues
};

File: kid-with-greatest-number-of-candies/kid-with-greatest-number-of-candies.py
class Solution(object):
    def kidsWithCandies(self, candies, extraCandies):
        """
        :type candies: List[int]
        :type extraCandies: int
        :rtype: List[bool]
        """ 
        truthValues = []
        maxCandies = max(candies)
        for candy in candies:
            if candy + extraCandies >= maxCandies:
                truthValues.append(True)
            else:
                truthValues.append(False)

        return truthValues